base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -673,6 +694,143 @@ where
+        if !stp.is_collecting_single_signature() {
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "Sender protocol single round signature not being collected".to_string(),
+            ));
+        }
+        self.output_manager_service
+            .confirm_pending_transaction(tx_id)
+            .await
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+
+        // Prepare receiver part of the transaction
+
+        // Diffie-Hellman shared secret `k_Ob * K_Sb = K_Ob * k_Sb` results in a public key, which is converted to
+        // bytes to enable conversion into a private key to be used as the spending key
+        // TODO: Add a standardized Diffie-Hellman method to the tari_crypto library, then come back and use it here
+        let script_offset_private_key = stp
+            .get_recipient_script_offset_private_key(0)
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+        let spending_key = PrivateKey::from_bytes((script_offset_private_key.clone() * dest_pubkey.clone()).as_bytes())
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+
+        let sender_message = TransactionSenderMessage::new_single_round_message(stp.get_single_round_message()?);
+        let rtp = ReceiverTransactionProtocol::new(
+            sender_message,
+            PrivateKey::random(&mut OsRng),
+            spending_key,
+            OutputFeatures::default(),
+            &self.resources.factories,
+        );
+
+        let recipient_reply = rtp.get_signed_data()?.clone();
+
+        // Start finalizing
+
+        stp.add_single_recipient_info(recipient_reply, &self.resources.factories.range_proof)
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+        if !stp.is_finalizing() {
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "Sender protocol single round tx not being finalized".to_string(),
+            ));
+        }
+
+        // Finalize
+
+        stp.finalize(KernelFeatures::empty(), &self.resources.factories)
+            .map_err(|e| {
+                error!(
+                    target: LOG_TARGET,
+                    "Transaction (TxId: {}) could not be finalized. Failure error: {:?}", tx_id, e,
+                );
+                TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e))
+            })?;
+        if !stp.is_finalized() {
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "Sender protocol single round tx not finalized".to_string(),
+            ));
+        }
+        trace!(target: LOG_TARGET, "Finalized one-side transaction TxId: {}", tx_id);
+
+        let _ = self
```

Comment:
What's the reason for swallowing this error?

Reply:
This event being sent is important, but not critical to the protocol being successful. It could be that the receiver dropped a handle for some reason.

_From the Tokio documentation:_
A return value of Ok does not mean that the sent value will be observed by all or any of the active Receiver handles. Receiver handles may be dropped before receiving the sent message. A return value of Err does not mean that future calls to send will fail.

Reply:
Anything like this that is out of the ordinary should have a comment stating why it can be swallowed, so that new devs don't copy this code thinking it's the norm

Reply:
Agreed, added a proper comment